### PR TITLE
Improve EditorPlugin state documentation

### PR DESCRIPTION
**Pull request by KoBeWi**
This PR provides more details for `_get_state`, `_get_window_layout`, `_set_state` and `_set_window_layout` methods.

### CODE REVIEW

File: doc/classes/EditorPlugin.xml [hunk 1]
```diff
@@ -261,14 +261,28 @@
...
+				    return state
+				[/codeblock]
 			</description>
 		</method>
 		<method name="_get_window_layout" qualifiers="virtual">
 			<return type="void" />
 			<argument index="0" name="configuration" type="ConfigFile" />
 			<description>
-				Gets the GUI layout of the plugin. This is used to save the project's editor layout when [method queue_save_layout] is called or the editor layout was changed(For example changing the position of a dock).
+				Override this method to provide the GUI layout of the plugin or just any data you want to be stored. This is used to save the project's editor layout when [method queue_save_layout] is called or the editor layout was changed (for example changing the position of a dock). The data is stored in the [code]editor_layout.cfg[/code] file inside editor metadata directory.
```
Comment (Calinou) [inline doc/classes/EditorPlugin.xml:279]:
```suggestion
				Override this method to provide the GUI layout of the plugin or any other data you want to be stored. This is used to save the project's editor layout when [method queue_save_layout] is called or the editor layout was changed (for example changing the position of a dock). The data is stored in the [code]editor_layout.cfg[/code] file in the editor metadata directory.
```

File: doc/classes/EditorPlugin.xml [hunk 2]
```diff
@@ -261,14 +261,28 @@
 		<method name="_get_state" qualifiers="virtual const">
 			<return type="Dictionary" />
 			<description>
-				Gets the state of your plugin editor. This is used when saving the scene (so state is kept when opening it again) and for switching tabs (so state can be restored when the tab returns).
+				Override this method to provide a state data you want want to be saved, like view position, grid settings, folding etc. This is used when saving the scene (so state is kept when opening it again) and for switching tabs (so state can be restored when the tab returns). This data is automatically saved for each scene in an [code]editstate[/code] file inside editor metadata folder. If you want to store global (scene-independent) editor data for your plugin, you can use [method _get_window_layout] instead.
```
Comment (Calinou) [inline doc/classes/EditorPlugin.xml:264]:
```suggestion
				Override this method to provide a state data you want to be saved, like view position, grid settings, folding, etc. This is used when saving the scene (so state is kept when opening it again) and for switching tabs (so state can be restored when the tab returns). This data is automatically saved for each scene in an [code]editstate[/code] file in the editor metadata folder. If you want to store global (scene-independent) editor data for your plugin, you can use [method _get_window_layout] instead.
```

**Review by Calinou — DISMISSED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**